core/src/main/java/org/elasticsearch/Version.java

```diff
@@ -95,18 +95,18 @@
     public static final Version V_5_0_0 = new Version(V_5_0_0_ID, org.apache.lucene.util.Version.LUCENE_6_2_0);
     public static final int V_5_0_1_ID = 5000199;
     public static final Version V_5_0_1 = new Version(V_5_0_1_ID, org.apache.lucene.util.Version.LUCENE_6_2_1);
-    public static final int V_6_0_0_alpha1_ID = 6000001;
-    public static final Version V_6_0_0_alpha1 = new Version(V_6_0_0_alpha1_ID, org.apache.lucene.util.Version.LUCENE_6_3_0);
-    public static final Version CURRENT = V_6_0_0_alpha1;
-
-    /* NOTE: don't add unreleased version to this list except of the version assigned to CURRENT.
-     * If you need a version that doesn't exist here for instance V_5_1_0 then go and create such a version
-     * as a constant where you need it:
-     * <pre>
-     *   public static final Version V_5_1_0_UNRELEASED = Version.fromId(5010099);
-     * </pre>
-     * Then go to VersionsTest.java and add a test for this constant VersionTests#testUnknownVersions().
-     * This is particularly useful if you are building a feature that needs a BWC layer for this unreleased version etc.*/
```

Comment:
This long comment makes me wonder what the cons are for having unreleased versions in this class. the way we deal with them with this PR looks good to me though.